rust/lance-table/src/format/manifest.rs

```diff
@@ -160,6 +167,18 @@ impl Manifest {
         self.timestamp_nanos = nanos;
     }
 
+    /// Set the `table_metadata` from a metadata HashMap
+    pub fn set_metadata(&mut self, metadata: HashMap<String, String>) {
```

Comment:
```suggestion
    pub fn set_metadata(&mut self, metadata: impl IntoIterator<Item = (String, String)>) {
```

Reply:
Addressed. I've also changed `lance::dataset::Dataset.set_table_metadata` to use same argument type - is that OK?